File: Gestion Coach Client/src/Entite/Utilisateur.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Entite;

import java.sql.Date;

/**
 *
 * @author asus
 */
public class Utilisateur {
    private int id;
    private String surnom;
    private String prenom;
    private String nom;
    private String numeroDeTelephone;
    private String email;
    private String password;
    private String dateDeNaissance;
    private String sexe;
    private String role;
    private int nbrEvaluation;
    private float evaluationCoach;
    
    
    
   
   
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getSurnom() {
        return surnom;
    }

    public void setSurnom(String surnom) {
        this.surnom = surnom;
    }

    public String getPrenom() {
        return prenom;
    }

    public void setPrenom(String prenom) {
        this.prenom = prenom;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getNumeroDeTelephone() {
        return numeroDeTelephone;
    }

    public void setNumeroDeTelephone(String numeroDeTelephone) {
        this.numeroDeTelephone = numeroDeTelephone;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getDateDeNaissance() {
        return dateDeNaissance;
    }

    public void setDateDeNaissance(String dateDeNaissance) {
        this.dateDeNaissance = dateDeNaissance;
    }

 

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

   

    public String getSexe() {
        return sexe;
    }

    public void setSexe(String sexe) {
        this.sexe = sexe;
    }

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }

    public int getNbrEvaluation() {
        return nbrEvaluation;
    }

    public void setNbrEvaluation(int nbrEvaluation) {
        this.nbrEvaluation = nbrEvaluation;
    }

    public float getEvaluationCoach() {
        return evaluationCoach;
    }

    public void setEvaluationCoach(float evaluationCoach) {
        this.evaluationCoach = evaluationCoach;
    }

    

    public Utilisateur() {
    }

    public Utilisateur(int id) {
        this.id = id;
    }

    public Utilisateur(int id, String surnom, String email, String role) {
        this.id = id;
        this.surnom = surnom;
        this.email = email;
        this.role = role;
    }

    public Utilisateur(int nbrEvaluation, float evaluationCoach) {
        this.nbrEvaluation = nbrEvaluation;
        this.evaluationCoach = evaluationCoach;
    }

    public Utilisateur(String surnom, String prenom, String nom, String numeroDeTelephone, String email, String password, String dateDeNaissance, String sexe, String role, int nbrEvaluation, float evaluationCoach) {
        this.surnom = surnom;
        this.prenom = prenom;
        this.nom = nom;
        this.numeroDeTelephone = numeroDeTelephone;
        this.email = email;
        this.password = password;
        this.dateDeNaissance = dateDeNaissance;
        this.sexe = sexe;
        this.role = role;
        this.nbrEvaluation = nbrEvaluation;
        this.evaluationCoach = evaluationCoach;
    }

    public Utilisateur(int id, String surnom, String prenom, String nom, String numeroDeTelephone, String email, String password, String dateDeNaissance, String sexe, String role, int nbrEvaluation, float evaluationCoach) {
        this.id = id;
        this.surnom = surnom;
        this.prenom = prenom;
        this.nom = nom;
        this.numeroDeTelephone = numeroDeTelephone;
        this.email = email;
        this.password = password;
        this.dateDeNaissance = dateDeNaissance;
        this.sexe = sexe;
        this.role = role;
        this.nbrEvaluation = nbrEvaluation;
        this.evaluationCoach = evaluationCoach;
    }

    

    
 
    public Utilisateur(int id, String surnom, String prenom, String nom, String email) {
        this.id = id;
        this.surnom = surnom;
        this.prenom = prenom;
        this.nom = nom;
        this.email = email;
    }
    
   
   
    
 public Utilisateur(String surnom) {
        this.surnom = surnom;
    }

    public Utilisateur(String surnom, String email, String role) {
        this.surnom = surnom;
        this.email = email;
        this.role = role;
    }

    public Utilisateur(String surnom, String prenom, String nom, String numeroDeTelephone, String email, float evaluationCoach) {
        this.surnom = surnom;
        this.prenom = prenom;
        this.nom = nom;
        this.numeroDeTelephone = numeroDeTelephone;
        this.email = email;
        this.evaluationCoach = evaluationCoach;
    }

    public Utilisateur(int id, String surnom, String prenom, String nom, String numeroDeTelephone, String email, String password, String dateDeNaissance, String sexe, String role) {
        this.id = id;
        this.surnom = surnom;
        this.prenom = prenom;
        this.nom = nom;
        this.numeroDeTelephone = numeroDeTelephone;
        this.email = email;
        this.password = password;
        this.dateDeNaissance = dateDeNaissance;
        this.sexe = sexe;
        this.role = role;
    }




    @Override
    public int hashCode() {
        int hash = 3;
        hash = 31 * hash + this.id;
        return hash;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final Utilisateur other = (Utilisateur) obj;
        if (this.id != other.id) {
            return false;
        }
        return true;
    }

    @Override
    public String toString() {
        return "Utilisateur{" + "id=" + id + ", surnom=" + surnom + ", prenom=" + prenom + ", nom=" + nom + ", numeroDeTelephone=" + numeroDeTelephone + ", email=" + email + ", password=" + password + ", dateDeNaissance=" + dateDeNaissance + ", sexe=" + sexe + ", role=" + role + ", nbrEvaluation=" + nbrEvaluation + ", evaluationCoach=" + evaluationCoach + '}';
    }

    
  
      
}

   

File: Gestion Cours-WEB/src/Entity/Utilisateur.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * Utilisateur
 *
 * @ORM\Table(name="utilisateur", uniqueConstraints={@ORM\UniqueConstraint(name="id", columns={"id"}), @ORM\UniqueConstraint(name="email", columns={"email"})})
 * @ORM\Entity
 */
class Utilisateur
{
    /**
     * @var int
     *
     * @ORM\Column(name="id", type="integer", nullable=false)


     */
    private $id;

    /**
     * @var string|null
     *
     * @ORM\Column(name="surnom", type="string", length=20, nullable=true)
     */
    private $surnom;

    /**
     * @var string|null
     *
     * @ORM\Column(name="nom", type="string", length=30, nullable=true)
     */
    private $nom;

    /**
     * @var string|null
     *
     * @ORM\Column(name="prenom", type="string", length=30, nullable=true)
     */
    private $prenom;

    /**
     * @var string|null
     *
     * @ORM\Column(name="email", type="string", length=30, nullable=true)
     * @ORM\Id
     */
    private $email;

    /**
     * @var string|null
     *
     * @ORM\Column(name="password", type="string", length=30, nullable=true)
     */
    private $password;

    /**
     * @var \DateTime|null
     *
     * @ORM\Column(name="DateDeNaissance", type="date", nullable=true)
     */
    private $datedenaissance;

    /**
     * @var string|null
     *
     * @ORM\Column(name="sexe", type="string", length=20, nullable=true)
     */
    private $sexe;

    /**
     * @var string|null
     *
     * @ORM\Column(name="role", type="string", length=30, nullable=true)
     */
    private $role;

    /**
     * @return int
     */
    public function getId(): ?int
    {
        return $this->id;
    }

    /**
     * @param int $id
     */
    public function setId(int $id): void
    {
        $this->id = $id;
    }

    /**
     * @return string|null
     */
    public function getSurnom(): ?string
    {
        return $this->surnom;
    }

    /**
     * @param string|null $surnom
     */
    public function setSurnom(?string $surnom): void
    {
        $this->surnom = $surnom;
    }

    /**
     * @return string|null
     */
    public function getNom(): ?string
    {
        return $this->nom;
    }

    /**
     * @param string|null $nom
     */
    public function setNom(?string $nom): void
    {
        $this->nom = $nom;
    }

    /**
     * @return string|null
     */
    public function getPrenom(): ?string
    {
        return $this->prenom;
    }

    /**
     * @param string|null $prenom
     */
    public function setPrenom(?string $prenom): void
    {
        $this->prenom = $prenom;
    }

    /**
     * @return string|null
     */
    public function getEmail(): ?string
    {
        return $this->email;
    }

    /**
     * @param string|null $email
     */
    public function setEmail(?string $email): void
    {
        $this->email = $email;
    }

    /**
     * @return string|null
     */
    public function getPassword(): ?string
    {
        return $this->password;
    }

    /**
     * @param string|null $password
     */
    public function setPassword(?string $password): void
    {
        $this->password = $password;
    }

    /**
     * @return \DateTime|null
     */
    public function getDatedenaissance(): ?\DateTime
    {
        return $this->datedenaissance;
    }

    /**
     * @param \DateTime|null $datedenaissance
     */
    public function setDatedenaissance(?\DateTime $datedenaissance): void
    {
        $this->datedenaissance = $datedenaissance;
    }

    /**
     * @return string|null
     */
    public function getSexe(): ?string
    {
        return $this->sexe;
    }

    /**
     * @param string|null $sexe
     */
    public function setSexe(?string $sexe): void
    {
        $this->sexe = $sexe;
    }

    /**
     * @return string|null
     */
    public function getRole(): ?string
    {
        return $this->role;
    }

    /**
     * @param string|null $role
     */
    public function setRole(?string $role): void
    {
        $this->role = $role;
    }

    public function __toString() {
        return $this->getEmail();
    }


}
